docs/javascript/imports.md

```diff
@@ -1,118 +1,194 @@
 # JavaScript: Imports
 
-You can load a library using an `import` statement. For example:
+You can load a library using an [`import` statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import). For example, to load [canvas-confetti](https://github.com/catdad/canvas-confetti) from npm:
 
 ```js echo
 import confetti from "npm:canvas-confetti";
 ```
 
-<div class="tip">Thanks to <a href="./reactivity">reactivity</a>, imports can live in a JavaScript block anywhere on the page. But by convention, imports are commonly placed at the top of source files for readability.</div>
+The code above imports the default export and is equivalent to:
 
-You can use an imported library anywhere on the page.
+```js run=false
+import {default as confetti} from "npm:canvas-confetti";
+```
+
+Depending on the package, you may want to import specific named exports, or to import everything as a namespace. For example:
+
+```js run=false
+import {rollup} from "npm:d3-array"; // a single named import
+import * as d3 from "npm:d3"; // a namespace import
```

Comment:
This shows two changes. I understand it makes sense to only import from d3-array when you need only rollup, but it makes the example more difficult to understand. (People might wonder how d3-array is different from d3, and how they could tell.)

Reply:
The point is “depending on the package” in the line above. It depends on what you need too, of course. But I wouldn’t recommend `import {rollup} from "npm:d3"` since you would pull in all 30 packages even though you just need d3-array. And I wouldn’t recommend `import * as d3 from "npm:d3-array"` because it gives the impression that all of D3 is available. So we could say `import * as array from "npm:d3-array"` here, but is that better? I can’t think of a situation where I would use that over specific named imports.

Reply:
maybe reorder and adapt the comments:

```suggestion
import * as d3 from "npm:d3"; // import all of D3 with a namespace import
import {rollup} from "npm:d3-array"; // a single named import from a specific module
```

Reply:
I ordered it to match the sentence above “you may want to import specific named exports, or to import everything as a namespace.” I can reorder that sentence too but I don’t understand why it’s easier to understand if the namespace is first? If anything I think importing the default export as shown in the canvas-confetti examples is closer to importing a namespace.

Reply:
I changed the namespace comment to `// import everything as a namespace`. Does that help?